tests/python/gaia-ui-tests/gaiatest/apps/music/regions/list_view.py

```diff
@@ -0,0 +1,65 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette_driver import expected, By, Wait
+
+from gaiatest.apps.base import Base, PageRegion
+from gaiatest.apps.music.regions.sublist_view import SublistView
+from gaiatest.apps.music.regions.player_view import PlayerView
+
+
+class ListView(Base):
+
+    _albums_view_locator = (By.CSS_SELECTOR, 'iframe.active[src*="/views/albums/index.html"]')
+    _artists_view_locator = (By.CSS_SELECTOR, 'iframe.active[src*="/views/artists/index.html"]')
+    _songs_view_locator = (By.CSS_SELECTOR, 'iframe.active[src*="/views/songs/index.html"]')
+    _playlists_view_locator = (By.CSS_SELECTOR, 'iframe.active[src*="/views/playlists/index.html"]')
+    _view_locator = (By.ID, 'list')
+    _list_item_locator = (By.CLASS_NAME, 'gfl-item')
+
+    def __init__(self, marionette, view_type):
+        Base.__init__(self, marionette)
+        if view_type is 'albums':
```

Comment:
How about breaking this List views into smaller classes? Something like:
``` python
class ListView(Base):
    def __init__(self, marionette):
         Base.__init__(self, marionette)
         self._wait_for_main_view_to_be displayed()

    def media():
        ....

    def get_view_by_type_locator(self, type)
        return By.CSS_SELECTOR, 'iframe.active[src*="/views/{}/index.html"]'.format(type)

class ArtistView(ListView):
    def _wait_for_main_view_to_be displayed(self):
       Wait(self.marionette).until(expected.element_displayed(*self.get_view_by_type_locator('artists')))

class AlbumView(ListView):
    ...
```

What do you think?